content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-safeql.mdx

```diff
@@ -0,0 +1,369 @@
+
+```yaml
+version: '3.6'
+services:
+  pgDB:
+    image: postgis/postgis:13-3.1-alpine
+    restart: always
+    ports:
+      - '5432:5432'
+    volumes:
+      - db_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_PASSWORD: password
+      POSTGRES_DB: geoexample
+volumes:
+  db_data:
+```
+
+</Admonition>
+
+Next, create a migration and execute a migration to enable the extension:
+
+```terminal
+npx prisma migrate dev --name add-postgis
+```
+
+For reference, the output of the migration file should look like the following:
+
+```sql file=migrations/TIMESTAP_add_postgis/migration.sql
+-- CreateExtension
+CREATE EXTENSION IF NOT EXISTS "postgis";
+```
+
+You can double-check that the migration has been applied by running `prisma migrate status`.
+
+## 2. Create a new model that uses a geographic data column
+
+Add a new model with a column with a `geography` data type once the migration is applied. For this guide, we'll use a model called `PointOfInterest`.
+
+```prisma
+model PointOfInterest {
+  id       Int                                   @id @default(autoincrement())
+  name     String
+  location Unsupported("geography(Point, 4326)")
+}
+```
+
+You'll notice that the `location` field uses an [`Unsupported`](/reference/prisma-schema-reference#unsupported) type. This means that we lose a lot of the benefits of Prisma when working with `PointOfInterest`. We'll be using [SafeQL](https://safeql.dev/) to fix this.
+
+Like before, create and execute a migration `prisma migrate dev` to create the `PointOfInterest` table in your database:
+
+```terminal
+npx prisma migrate dev --name add-poi
+```
+
+Again for reference, here is the output of the SQL migration file generated by Prisma Migrate:
+
+```sql file=migrations/TIMESTAP_add_poi/migration.sql
+-- CreateTable
+CREATE TABLE "PointOfInterest" (
+    "id" SERIAL NOT NULL,
+    "name" TEXT NOT NULL,
+    "location" geography(Point, 4326) NOT NULL,
+
+    CONSTRAINT "PointOfInterest_pkey" PRIMARY KEY ("id")
+);
+```
+
+## 3. Integrate SafeQL
+
+SafeQL is easily integrated with Prisma in order to lint `$queryRaw` and `$executeRaw` Prisma calls. You can reference [SafeQL's integration guide](https://safeql.dev/compatibility/prisma.html) or follow the steps below.
+
+### 3.1. Install the <inlinecode>@ts-safeql/eslint-plugin</inlinecode> npm package
+
+```terminal
+npm install -D @ts-safeql/eslint-plugin
+```
+
+This ESLint plugin is what will allow for queries to be linted.
+
+### 3.2. Add <inlinecode>@ts-safeql/eslint-plugin</inlinecode> to your ESLint plugins
+
+You should now add `@ts-safeql/eslint-plugin` to your list of ESLint plugins. In our example we are using an `.eslintrc.js` file, but this can be applied to any way that you [configure ESLint](https://eslint.org/docs/latest/use/configure/).
```

Comment:
```suggestion
Add `@ts-safeql/eslint-plugin` to your list of ESLint plugins. In our example we are using an `.eslintrc.js` file, but this can be applied to any way that you [configure ESLint](https://eslint.org/docs/latest/use/configure/).
```